src/main/java/com/jfrog/xray/client/impl/XrayClientBuilder.java

```diff
@@ -0,0 +1,38 @@
+package com.jfrog.xray.client.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.jfrog.build.api.util.NullLog;
+import org.jfrog.build.client.PreemptiveHttpClientBuilder;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+/**
+ * @author yahavi
+ */
+public class XrayClientBuilder extends PreemptiveHttpClientBuilder {
+
+    private static final String DEFAULT_USER_AGENT = "jfrog-xray-client/" + XrayClient.class.getPackage().getImplementationVersion();
+    private static final int CONNECTION_TIMEOUT_MILLISECONDS = (int) MINUTES.toMillis(1);
```

Comment:
Let's add a description here, stating this is a 1 minute timeout (just to make it clearer)